src/main/java/uk/gov/hmcts/probate/validator/PartnersAddedValidationRule.java

```diff
@@ -0,0 +1,36 @@
+package uk.gov.hmcts.probate.validator;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.probate.exception.BusinessValidationException;
+import uk.gov.hmcts.probate.model.ccd.raw.request.CaseData;
+import uk.gov.hmcts.probate.model.ccd.raw.request.CaseDetails;
+import uk.gov.hmcts.probate.service.BusinessValidationMessageRetriever;
+
+import java.util.Locale;
+
+@Component
+@RequiredArgsConstructor
+public class PartnersAddedValidationRule {
+
+    private final BusinessValidationMessageRetriever businessValidationMessageRetriever;
+    private static final String PARTNERS_NEEDED = "partnersNeeded";
+
+    public void validate(CaseDetails caseDetails) {
+
+        CaseData caseData = caseDetails.getData();
+        String[] args = {caseDetails.getId().toString()};
+        String userMessage = businessValidationMessageRetriever.getMessage(PARTNERS_NEEDED, args, Locale.UK);
+
+        if (caseData.getAnyOtherApplyingPartners() != null) {
+            if (!(caseData.getSolsSolicitorIsExec().matches("No")
```

Comment:
Could potentially have combined these into 1 if statement with an &&

Reply:
or a more succint method, and one that seems to be used quite a lot in the Java world, and in probate-back-office, is:
if (YES.equals(caseData.getSolsSolicitorIsExec()
    && YES.equals(caseData.getSolsSolicitorIsApplying())
    && NO.equals(caseData.getAnyOtherApplyingPartners())
....

Wouldn't need the outer if then

Reply:
Done.